### ISSUE TITLE
Pre-annotate video frames?

### ISSUE THREAD

**Issue by bbernhard** [bbernhard/imagemonkey-core#35]
I was playing a little bit with the idea of detecting object boundaries within video frames (idea originated in #32). 

Currently my implementation looks like this: 

* use image subtraction (weighted moving mean algorithm). 
* find contours on foreground mask
* cluster nearby contours by using DBSCAN algorithm
* find minimal bounding rectangle for all contours in cluster

Here is a GIF that shows the result (I used some random video from Youtube as input): 

![video_extraction_test](https://user-images.githubusercontent.com/5710631/31459148-96ac0dd2-aec2-11e7-89c2-9b66b65deec9.gif)

However I am not really happy with the result...it definitely got better after using the DBSCAN algorithm, but it's still not as good as I would have expected. Also tried a few different algorithms, but the "weighted moving mean" was creating the best results (at least for my test video). 

Anyone an idea on how to improve the results?

**Comment by dobkeratops** [comment]
interesting experiment.. i think its worth it.. you can always manually clean up.
one comment, maybe polygonal boundaries would be worthwhile (8 sided polls, whatever), so you don't get noise in the rotations from frame to frame

**Comment by bbernhard** [comment]
Thanks for your advice, really appreciated!

> maybe polygonal boundaries would be worthwhile (8 sided, whatever), so you don't get noise in rotations from frame to frame (it would localise the effect of the noise more)

Do you mean something like a convex hull?

**Comment by dobkeratops** [comment]
yes exactly a convex hull. maybe it's worth trying a bounding box clipped with diagonals, like this
![definitions and tools k-dop an 8-dop in two dimensions](https://user-images.githubusercontent.com/1120754/31522095-07451b24-afa4-11e7-8d44-f38fd962af57.jpg) .. or something more general. My theory is the aligned+diagonal version would be more stable between frames .. but I haven't tried anything like this

**Comment by bbernhard** [comment]
Coool, thanks :) Will definitely check that out!
